test/unittest/tests/extent_list.cxx

```diff
@@ -1,13 +1,393 @@
+#include <memory>
+
 #include <catch2/catch.hpp>
 
-uint32_t factorial(uint32_t num) {
-    return num > 1 ? factorial(num - 1) * num : 1;
+#include "wt_internal.h"
+
+struct ExtentWrapper {
+    ExtentWrapper(WT_EXT* raw)
+        : _raw(raw) {
+    }
+
+    ~ExtentWrapper() {
+        free(_raw);
+    }
+
+    WT_EXT *_raw;
+};
+
+struct ExtentListWrapper {
+    ExtentListWrapper() {
+    }
+
+    ~ExtentListWrapper() {
+    }
+
+    // unfortunately, some functions need raw 2D pointers, but that's not
+    // compatible with automatic memory management. _list is only for allocation
+    // bookkeeping - _raw_list can be rearranged however.
+    std::vector<std::unique_ptr<ExtentWrapper>> _list;
+    std::vector<WT_EXT*> _raw_list;
+};
+
+struct SizeWrapper {
+    SizeWrapper(WT_SIZE* raw)
+        : _raw(raw) {
+    }
+
+    ~SizeWrapper() {
+        free(_raw);
+    }
+
+    WT_SIZE *_raw;
+};
+
+struct SizeListWrapper {
+    SizeListWrapper() {
+    }
+
+    ~SizeListWrapper() {
+    }
+
+    // unfortunately, some functions need raw 2D pointers, but that's not
+    // compatible with automatic memory management. _list is only for allocation
+    // bookkeeping - _raw_list can be rearranged however.
+    std::vector<std::unique_ptr<SizeWrapper>> _list;
+    std::vector<WT_SIZE*> _raw_list;
+};
+
+std::unique_ptr<ExtentWrapper> create_new_ext() {
```

Comment:
I'm not quite sure what you mean by "that is generic", sorry - can you elaborate on this a bit?

Reply:
At the moment, the code can only create an extent list of a fixed size. I was thinking it might be useful to pull out the size calculation into a method that is seperate from the memory allocation code. Again, this could wait until we refactor the wrapper classes into seperate files.